src/App.js

```diff
@@ -151,29 +160,34 @@ export default function App() {
         <React.Fragment>
             <CssBaseline />
             <Navbar/>
-            <main>
-                {/* Hero unit */}
-                <div className={classes.heroContent}>
-                    <Container maxWidth="md">
-                            { Object.keys(supportedLanguages).length > 0 &&
-                              <LanguageBar 
-                                supportedLanguages = {supportedLanguages}
-                                inputLanguage={inputLanguage} 
-                                outputLanguage={outputLanguage} 
-                                handleInputLanguageChange= {handleInputLanguageChange} 
-                                handleOutputLanguageChange= {handleOutputLanguageChange} 
+            <div>
+              {loading 
```

Comment:
I am a react noob so this might be bad advice. I think it would be a little cleaner if you had a loading component and a component for everything else and this file just switched between the two component based on if loading.

Reply:
+1

Reply:
You can also return from the function early:
```ts
if (loading) {
  return (
    <div className={classes.loadingPage}><CircularProgress size="100px"/></div>
  );
}

return (
  <React.Fragment>
    ...
```